functions/Get-DbaDbRoleMember.ps1

```diff
@@ -144,14 +144,14 @@ function Get-DbaDbRoleMember {
                         }
 
                         if ($user) {
-                            Add-Member -Force -InputObject $user -MemberType 'NoteProperty' -Name 'ComputerName' -Value $server.ComputerName
-                            Add-Member -Force -InputObject $user -MemberType 'NoteProperty' -Name 'InstanceName' -Value $server.ServiceName
-                            Add-Member -Force -InputObject $user -MemberType 'NoteProperty' -Name 'SqlInstance' -Value $server.DomainInstanceName
-                            Add-Member -Force -InputObject $user -MemberType 'NoteProperty' -Name 'Database' -Value $db.Name
-                            Add-Member -Force -InputObject $user -MemberType 'NoteProperty' -Name 'Role' -Value $dbRole.Name
-                            Add-Member -Force -InputObject $user -MemberType 'NoteProperty' -Name 'UserName' -Value $user.Name
-
-                            Select-DefaultView -InputObject $user -Property 'ComputerName', 'InstanceName', 'SqlInstance', 'Database', 'Role', 'UserName', 'Login'
+                            $user | Add-Member -Force -MemberType 'NoteProperty' -Name 'ComputerName' -Value $server.ComputerName
```

Comment:
We don't use the piping option in any of the commands because it is slower. Any reason you chose to do it on this one?

Reply:
yes, when there is a missing foreach () loop. This has multiple $user so it needs to foreach in the pipe. I think. I'm seeing REALLY weird stuff.

Look at this

![image](https://user-images.githubusercontent.com/8278033/46572946-8e9d1b80-c98e-11e8-9694-ba38f79124f1.png)

Reply:
Your `OutVariable roles` needs to be prior to the pipe and filter.